gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineReleaseCompatibility.java

```diff
@@ -56,8 +56,7 @@ private ReleaseFlagProvider(JavaCompile javaCompile) {
 
         @Override
         public Iterable<String> asArguments() {
-            JavaVersion jdkVersion =
-                    JavaVersion.toVersion(javaCompile.getToolChain().getVersion());
```

Comment:
The `JavaCompile#getToolChain` method got removed and I am not sure what a good replacement is.

I replaced it with `targetCompat` for now but they are not equivalent.

Reply:
Does something like:

```java
Optional<JavaVersion> taskTarget = Optional.ofNullable(javaCompile
                .getProject()
                .getExtensions()
                .getByType(JavaPluginExtension.class)
                .getToolchain()
                .getLanguageVersion()
                .getOrNull())
        .map(JavaLanguageVersion::asInt)
        .map(JavaVersion::toVersion);
```

replace it exactly? Not sure if it's an exact replacement, but tests seem to pass at least.

Reply:
I think `JavaPluginExtension#getToolchain` is just always empty unless your explicitly configure a toolchain in your build.gradle

Reply:
I think what we can do is use `getToolchain().getLanguageVersion().orElseGet(JavaVersion::current)`. In this case we would use the toolchain jdk version if specified and otherwise fall back to the global java version.

Only thing is that the new `getToolchain` methods only exists since Gradle 6.7. We can either use Gradle 6.7 as the new minimum required Gradle version for baseline or do something like this again:
```
if (Gradle 6.7+) { get new toolchain through reflection }
else { get old toolchain through reflection }
```